presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -114,10 +114,9 @@
 public class SemiTransactionalHiveMetastore
 {
     private static final Logger log = Logger.get(SemiTransactionalHiveMetastore.class);
-    private static final int PARTITION_COMMIT_BATCH_SIZE = 8;
     private static final int MAX_LAST_DATA_COMMIT_TIME_ENTRY_PER_TABLE = 100;
     private static final int MAX_LAST_DATA_COMMIT_TIME_ENTRY_PER_TRANSACTION = 10_000;
-
+    private static final int PARTITION_COMMIT_BATCH_SIZE = 3000;
```

Comment:
Initially the high bound was set for performance testing but it could end up sending all partitionadd info in a single thrift call which could potentially cause metastore timeout error. So it's lowered to 3000 for now for sake of improving how memory is used in the metastore and to avoid client read timeout exceptions.

I believe for glue catalog it's safe with high bound since `BatchCreatePartitionRequest` is sent to glue metastore with 100 partition in a batch. I think the concern is for HMS and other metastores. 

I am open to make a reasonable change to define the value of this threshold.

Reply:
Yeah, for Glue it's capping it to 100. My only concern is about HMS thinking 3000 could be an overwhelming number for maybe smaller HMS instances

Reply:
ok it makes sense to align the limit to 100